src/courtroom.cpp

```diff
@@ -1565,59 +1636,115 @@ void Courtroom::handle_chatmessage(QStringList *p_contents)
+   qDebug() << "MAIN CHR NAME: " <<  m_chatmessage[CHAR_NAME];
+   QString tmpmsg = "";
+   chatlogpiece* temp = new chatlogpiece(ao_app->get_showname(char_list.at(f_char_id).name), f_showname, ": " + m_chatmessage[MESSAGE], false, m_chatmessage[TEXT_COLOR].toInt());
+
+   ic_chatlog_history.append(*temp);
 
   while(ic_chatlog_history.size() > log_maximum_blocks && log_maximum_blocks > 0)
   {
     ic_chatlog_history.removeFirst();
   }
-
-  append_ic_text(": " + m_chatmessage[MESSAGE], f_showname);
+  refresh_iclog(true);
+//HERE
+  if (f_showname == "")
+        f_showname = m_chatmessage[CHAR_NAME];
+  if(!mirror_iclog)
+    append_ic_text(": " + m_chatmessage[MESSAGE], f_showname,m_chatmessage[text_color].toInt());
 
   previous_ic_message = f_message;
-
-  int objection_mod = m_chatmessage[OBJECTION_MOD].toInt();
-  QString f_char = m_chatmessage[CHAR_NAME];
+  bool ok;
+  int objection_mod = m_chatmessage[OBJECTION_MOD].toInt(&ok, 10); //checks if its a custom obj.
+  objection_custom = ""; //added this for safety reasons
+  if(!ok && m_chatmessage[OBJECTION_MOD].contains("4&"))
+  {
+      objection_mod = 4;
+      objection_custom = m_chatmessage[OBJECTION_MOD].split("4&")[1]; //takes the name of custom objection.
+  }
+  QString f_char = char_name;
+  if(shown)
+    f_char = m_chatmessage[CHAR_NAME];
   QString f_custom_theme = ao_app->get_char_shouts(f_char);
-
+  ui_vp_message->clear();
+  ui_vp_chatbox->hide();
   //if an objection is used
   if (objection_mod <= 4 && objection_mod >= 1)
   {
+
     switch (objection_mod)
     {
     case 1:
-      ui_vp_objection->play("holdit", f_char, f_custom_theme);
-      objection_player->play("holdit.wav", f_char, f_custom_theme);
+      ui_vp_objection->play("holdit", f_char, f_custom_theme, shout_stay_time);
+      objection_player->play("holdit.wav", f_char, f_custom_theme); //Note: if sound breaks, add .wav back in the objection_player
       break;
     case 2:
-      ui_vp_objection->play("objection", f_char, f_custom_theme);
+      ui_vp_objection->play("objection", f_char, f_custom_theme, shout_stay_time);
       objection_player->play("objection.wav", f_char, f_custom_theme);
       if(ao_app->get_objectmusic())
           music_player->kill_loop();
       break;
     case 3:
-      ui_vp_objection->play("takethat", f_char, f_custom_theme);
+      ui_vp_objection->play("takethat", f_char, f_custom_theme, shout_stay_time);
       objection_player->play("takethat.wav", f_char, f_custom_theme);
       break;
     //case 4 is AO2 only
     case 4:
-      ui_vp_objection->play("custom", f_char, f_custom_theme);
-      objection_player->play("custom.wav", f_char, f_custom_theme);
+      if(objection_custom != "")
+      {
+          ui_vp_objection->play("custom_objections/" + objection_custom, f_char, f_custom_theme, shout_stay_time);
+          objection_player->play("custom_objections/" + objection_custom.split('.')[0] + ".wav", f_char, f_custom_theme);
```

Comment:
```suggestion
          objection_player->play("custom_objections/" + objection_custom.split('.')[0] + ".wav", f_char, f_custom_theme);
          objection_custom = "";
```
Set this back to empty string once you've played it - otherwise it will just play again if you try to left-click the button.

Reply:
Added it in the options for people to choose whether the custom objections should be maintened/cleared

Reply:
Why would you do that? This is clearly a logic error - it was even reported as a bug: #120

Reply:
Making this configurable is feature creep. Decide on a specific behavior and stick to it.

Reply:
I don't see it as that much optional for future creep since its hidden in the settings but whatever, I'll stick to the one-shot type.